### DISCUSSION TITLE
InstagramLoginRedirectError

### DISCUSSION THREAD

**Discussion by 42musaev** [chris-greening/instascrape#66] [Q&A]
Hello. Thank you for the project.
I try to make queries from the local machine - everything works.
But when I try to do it from vps I get InstagramLoginRedirectError. I have tried different vps.
I've read the FAQ, it says that you have to simulate a normal user. Please give me recommendations.
I am use Ubuntu 20.04 VPS

**Comment by chris-greening** [comment]
Hello!

Unfortunately I have had a very similar problem that is not currently solvable with `instascrape` alone. For `instascrape` to work, you have to manually login to Instagram from a browser so that your computer gets the necessary cookies required to make requests to Instagram. That being said, you'll have to figure out a way to login to Instagram from your remote server for the library to work properly. 

Cheers,
Chris

**Reply by 42musaev** [reply]
I managed to enter Instagram through requests by sending the necessary parameters, I received the authorized status, but I do not understand how this can be used with your library

**Reply by chris-greening** [reply]
That's actually really interesting, would you mind sharing your implementation of the login script? I tried a few times to bypass the login via `requests` but wasn't able to find any success, I'd love to see how you did it!

**Reply by 42musaev** [reply]
Just listed all the headers from the browser

```import requests

headers = {
    'accept': '*/*',
    'accept-encoding': 'gzip, deflate, br',
    'accept-language': 'ru-RU,ru;q=0.9,en-US;q=0.8,en;q=0.7',
    'content-length': '356',
    'content-type': 'application/x-www-form-urlencoded',
    'cookie': 'ig_did=F18315B7-7DEA-4BFA-8F38-8AF82FBC57AB; mid=X_RngAAEAAEZAQrrnS-93FPIu88w; ig_nrcb=1; shbid=738; shbts=1609929155.6013803; csrftoken=qeEl5bIs1U6OLivmamDgbFI0uSWj6LJ4',
    'origin': 'https://www.instagram.com',
    'referer': 'https://www.instagram.com/accounts/login/?next=%2F42musaev%2F&source=desktop_nav',
    'sec-ch-ua': '"Google Chrome";v="87", " Not;A Brand";v="99", "Chromium";v="87"',
    'sec-ch-ua-mobile': '?0',
    'sec-fetch-dest': 'empty',
    'sec-fetch-mode': 'cors',
    'sec-fetch-site': 'same-origin',
    'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36',
    'x-csrftoken': 'qeEl5bIs1U6OLivmamDgbFI0uSWj6LJ4',
    'x-ig-app-id': '936619743392459',
    'x-ig-www-claim': '0',
    'x-instagram-ajax': 'f9c4ed4055bb',
    'x-requested-with': 'XMLHttpRequest',
}


def login(user, password):
    data = {
        'username': user,
        'enc_password': '#PWD_INSTAGRAM_BROWSER:0:&:' + password,
    }

    response = requests.post('https://www.instagram.com/accounts/login/ajax/', headers=headers, data=data)
    return response
```

[image *click*](https://imgur.com/a/S99R2KS)
